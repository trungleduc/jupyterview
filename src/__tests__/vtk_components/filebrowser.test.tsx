import RemoteFileBrowser from "../../vtk_components/filebrowser";
import { mount, shallow, render, ShallowWrapper } from "enzyme";

import React from "react";
import {mock_send_msg, MockModel} from "../../test_setup/utils"

describe("Test RemoteFileBrowser component", () => {
  let component: ShallowWrapper
  let model : MockModel
  beforeAll(() => {
    model = new MockModel()
    model.set("root_data", [])
    component = shallow(<RemoteFileBrowser send_msg = {mock_send_msg} model = {model as any} browserRef = {null} />);
  })

  it("should render correctly with mock props", () => {
    expect(component).toMatchSnapshot();
  });

  it("should have correct initial state", () => {  
    expect(component.state("files")).toEqual([])
  });

});
