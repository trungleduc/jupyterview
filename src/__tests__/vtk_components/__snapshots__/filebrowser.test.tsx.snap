// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RemoteFileBrowser component should render correctly with mock props 1`] = `
<div
  className="file-browser"
  style={
    Object {
      "maxHeight": "50vh",
      "overflow": "auto",
      "width": "100%",
    }
  }
>
  <DragDropContext(t)
    detailRenderer={[Function]}
    files={Array []}
    icons={
      Object {
        "Archive": <i
          aria-hidden="true"
          className="fa fa-file-archive-o"
        />,
        "Audio": <i
          aria-hidden="true"
          className="fa fa-file-audio-o"
        />,
        "Delete": <i
          aria-hidden="true"
          className="fa fa-trash-o"
        />,
        "Download": <i
          aria-hidden="true"
          className="fa fa-download"
        />,
        "Excel": <i
          aria-hidden="true"
          className="fa fa-file-excel-o"
        />,
        "File": <i
          aria-hidden="true"
          className="fa fa-file-o"
        />,
        "Folder": <i
          aria-hidden="true"
          className="fa fa-folder-o"
        />,
        "FolderOpen": <i
          aria-hidden="true"
          className="fa fa-folder-open-o"
        />,
        "Image": <i
          aria-hidden="true"
          className="fa fa-file-image-o"
        />,
        "Loading": <i
          aria-hidden="true"
          className="fa fa-circle-o-notch fa-spin"
        />,
        "PDF": <i
          aria-hidden="true"
          className="fa fa-file-pdf-o"
        />,
        "PowerPoint": <i
          aria-hidden="true"
          className="fa fa-file-powerpoint-o"
        />,
        "Rename": <i
          aria-hidden="true"
          className="fa fa-i-cursor"
        />,
        "Text": <i
          aria-hidden="true"
          className="fa fa-file-text-o"
        />,
        "Video": <i
          aria-hidden="true"
          className="fa fa-file-video-o"
        />,
        "Word": <i
          aria-hidden="true"
          className="fa fa-file-word-o"
        />,
      }
    }
  />
</div>
`;
